### PR TITLE
Jc 1269.delete own starting post

### PR DESCRIPTION
**Pull request by artexnet**
This Pull Request contains a fix for a major issue JC-1269. Please review and merge the changes. Would appreciate your feedback since this is my first contribution to JTalks.

### CODE REVIEW

**Comment by ctapobep** [comment]
@artexnet could you please squash the commits into 1? Also, please follow the [commit message format](https://github.com/jtalks-org/jcommune/blob/master/docs/commit-conventions.md).

@mixas27, would you be able to review the change?

**Comment by artexnet** [comment]
@ctapobep I'm getting an error trying to squash the commits, I'm probably doing something wrong. Should you cancel this pull request, or you can process this "as is"? The second commit is just an update from upstream (i.e your) repository, so it does not contain anything you should validate additionally.

**Comment by ctapobep** [comment]
Squashing is extremely simple, see this answer on SO: http://stackoverflow.com/a/5201642/886697

We don't want to pollute the history with this kind of unnecessary commits/merges.

**Comment by mixas27** [comment]
@artexnet, don't forget to add some unit tests

File: jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java [hunk 1]
```diff
@@ -785,4 +785,18 @@ public Post getDraftForUser(JCUser user) {
...
     }
+
+	/**
+	 * Checks if this topic contains only posts added by topic Owner.
+	 *
+	 * @return <code>true</code> if condition is followed, otherwise <code>false</code>
+	 */
+	public boolean isContainsOwnerPostsOnly() {
+		for (Post post : getDisplayedPosts()) {
+			if (post.getUserCreated().getId() != topicStarter.getId()) {
```
Comment (mixas27) [inline jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java]: Yuo can compare users with equals method.

File: jcommune-view/jcommune-web-view/src/main/webapp/WEB-INF/jsp/topic/postList.jsp [hunk 2]
```diff
@@ -212,7 +212,7 @@
                 <jtalks:hasPermission targetId="${topic.branch.id}" targetType="BRANCH"
                                       permission="BranchPermission.DELETE_OWN_POSTS">
                   <c:set var="isDeleteButtonAvailable"
-                         value='${userId == post.userCreated.id && postsPage.numberOfElements == 1}'/>
+                         value='${(userId == post.userCreated.id && topic.displayedPostsCount == 1) || topic.containsOwnerPostsOnly}'/>
```
Comment (mixas27) [inline jcommune-view/jcommune-web-view/src/main/webapp/WEB-INF/jsp/topic/postList.jsp]: `topic.containsOwnerPostsOnly` should be enough here

File: jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalTopicModificationService.java [hunk 3]
```diff
@@ -326,12 +326,12 @@ private void subscribeOnTopicIfNotificationsEnabled(Topic topic, JCUser currentU
      */
     @PreAuthorize("(hasPermission(#topic.branch.id, 'BRANCH', 'BranchPermission.DELETE_OWN_POSTS') and " +
             "#topic.topicStarter.username == principal.username and " +
-            "#topic.postCount == 1) or " +
+            "#topic.displayedPostsCount == 1) or " +
```
Comment (mixas27) [inline jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalTopicModificationService.java]: `topic.containsOwnerPostsOnly` should be used here

**Comment by artexnet** [comment]
Guys, sorry for the delay. I'm closing this pull request and will start a new one (that contains one commit). The squash/rebase I tried to do gives an error, I'm not sure about the reason, maybe it's the "merge" commits presence or I'm doing smth. wrong, however it didn't work.

**Comment by ctapobep** [comment]
You probably tried the squash via rebase.. And I gave a link specifically on the reset command.

**Comment by artexnet** [comment]
I have tried both, are you available for a quick guidance? (skype/hangout)

**Comment by ctapobep** [comment]
Yap, ping me in skype: stanislav.bashkirtsev.

**Pull request closed without merging**